### PR TITLE
Suggested resource string changes

### PR DESCRIPTION
**Pull request by lgolding**
@michaelcfanning

### CODE REVIEW

File: src/BinSkim.Rules/RuleResources.resx [hunk 1]
```diff
@@ -118,19 +118,19 @@
...
+    <value>'{0}' is a 64-bit image with a preferred base address below the 4GB boundary. Having a preferred base address below this boundary triggers a compatibility mode in Address Space Layout Randomization (ASLR) on recent versions of Windows that reduces the number of locations to which ASLR may relocate the binary. This reduces the effectiveness of ASLR at mitigating memory corruption vulnerabilities. To resolve this issue, either use the default preferred base address by removing any uses of /baseaddress from compiler command lines, or /BASE from linker command lines (recommended), or configure your program to start at a base address above 4GB when compiled for 64 bit platforms (by changing the constant passed to /baseaddress or /BASE). Note that if you choose to continue using a custom preferred base address, you will need to make this modification only for 64-bit builds, as base addresses above 4GB are not valid for 32-bit binaries.</value>
```
Comment (lgolding) [inline src/BinSkim.Rules/RuleResources.resx]: GitHub doesn't always highlight the inline differences, it seems, even though it correctly detects that the line has changed. In this case I changed `/baseaddress / /BASE` to `/baseaddress or /BASE

File: src/BinSkim.Rules/RuleResources.resx [hunk 2]
```diff
@@ -118,19 +118,19 @@
...
+    <value>64-bit images should have a preferred base address above the 4GB boundary to prevent triggering an Address Space Layout Randomization (ASLR) compatibility mode that decreases security. ASLR compatibility mode reduces the number of locations to which ASLR may relocate the binary, reducing its effectiveness at mitigating memory corruption vulnerabilities. To resolve this issue, either use the default preferred base address by removing any uses of /baseaddress from compiler command lines, or /BASE from linker command lines (recommended), or configure your program to start at a base address above 4GB when compiled for 64 bit platforms (by changing the constant passed to /baseaddress or /BASE). Note that if you choose to continue using a custom preferred base address, you will need to make this modification only for 64-bit builds, as base addresses above 4GB are not valid for 32-bit binaries.</value>
```
Comment (lgolding) [inline src/BinSkim.Rules/RuleResources.resx]: Same here.

File: src/BinSkim.Rules/RuleResources.resx [hunk 3]
```diff
@@ -151,10 +151,10 @@
...
   </data>
   <data name="BA2006_Error" xml:space="preserve">
-    <value>'{0}' was compiled with one or more modules which were not built using minimum required tool versions (compiler version {1}, linker version {2}). More recent tool chains contain mitigations that make it more difficult for an attacker to exploit vulnerabilities in programs they produce. To resolve this issue, compile and/or link your binary with more recent tools. If you are servicing a product where the tool chain cannot be modified (e.g. producing a hotfix for an already shipped version) ignore this warning.
+    <value>'{0}' was compiled with one or more modules which were not built using minimum required tool versions (compiler version {1}, linker version {2}). More recent tools contain mitigations that make it more difficult for an attacker to exploit vulnerabilities in programs they produce. To resolve this issue, compile and/or link your binary with more recent tools. If you are servicing a product where the tool chain cannot be modified (e.g. producing a hotfix for an already shipped version) ignore this warning.
```
Comment (lgolding) [inline src/BinSkim.Rules/RuleResources.resx]: I changed "tool chain" to "tools". Except in the last sentence, where it just seemed better, but that is the only place in the entire resource file where I did not make that change, and I could be persuaded to change it here too.

File: src/BinSkim.Rules/RuleResources.resx [hunk 4]
```diff
@@ -213,10 +213,10 @@ Modules triggering this check were:
     <value>'{0}' has the imports section marked executable. Because the loader will always mark the imports section as writable, it is important to mark this section as non-executable, so that an attacker cannot place shellcode here. To resolve this issue, ensure that your program does not mark the imports section as executable. Look for uses of /SECTION or /MERGE on the linker command line, or #pragma segment in source code, which change the imports section to be executable, or which merge the ".rdata" segment into an executable section.</value>
   </data>
   <data name="BA2010_Pass" xml:space="preserve">
-    <value>'{0}' does not have an imports section that is marked as executable, helping to prevent the exploitability of code vulnerabilities.</value>
+    <value>'{0}' does not have an imports section that is marked as executable, helping to prevent the exploitation of code vulnerabilities.</value>
```
Comment (lgolding) [inline src/BinSkim.Rules/RuleResources.resx]: I think it should either be "prevent the exploitation of" (which is what I chose), or "reduce the exploitability of", but not "prevent the exploitability of".

**Comment by michaelcfanning** [comment]
LGTM looks great, nice change.

File: src/BinSkim.Rules/RuleResources.resx [hunk 5]
```diff
@@ -228,7 +228,7 @@ Modules triggering this check were:
...
+    <value>Application code should not interfere with the stack protector. The stack protector (/GS) is a security feature of the compiler which makes it more difficult to exploit stack buffer overflow memory corruption vulnerabilities. The stack protector relies on a random number, called the "security cookie", to detect these buffer overflows. This 'cookie' is statically linked with your binary from a Visual C++ library in the form of the symbol __security_cookie. On recent Windows versions, the loader looks for the statically linked value of this cookie, and initializes the cookie with a far better source of entropy -- the system's secure random number generator -- rather than the limited random number generator available early in the C runtime startup code. When this symbol is not the default value, the additional entropy is not injected by the operating system, reducing the effectiveness of the stack protector. To resolve this issue, ensure that your code does not reference or create a symbol named __security_cookie or __security_cookie_complement.</value>
```
Comment (lgolding) [inline src/BinSkim.Rules/RuleResources.resx]: I removed the word "magic".

**Pull request merged**